src/visualization.rs

```diff
@@ -0,0 +1,139 @@
+use crate::issues;
+use geojson::Value::Point;
+use geojson::{Feature, FeatureCollection};
+use gtfs_structures::{Gtfs, ObjectType};
+use serde_json::{to_value, Map};
+use std::sync::Arc;
+
+pub fn generate_issue_visualization(
+    issue: &issues::Issue,
+    gtfs: &Gtfs,
+) -> Option<FeatureCollection> {
+    match issue.object_type {
+        Some(ObjectType::Stop) => {
+            let stop_id = issue.object_id.clone();
+            let related_stop_ids = get_related_stop_ids(issue);
+
+            // a vec containing the stop_id and the related stop ids features
+            let stop_features: Vec<_> = [stop_id.clone()]
+                .iter()
+                .chain(related_stop_ids.iter())
+                .map(|stop_id| geojson_feature_point(&stop_id, gtfs))
+                .flatten()
+                .collect();
+
+            let line_string_features: Vec<_> = related_stop_ids
+                .iter()
+                .map(|related_stop| {
+                    geojson_feature_line_string(&stop_id, related_stop, gtfs, issue)
+                })
+                .flatten()
+                .collect();
+
+            let features = stop_features
+                .into_iter()
+                .chain(line_string_features.into_iter())
+                .collect();
+
+            let feature_collection = FeatureCollection {
+                bbox: None,
+                features: features,
+                foreign_members: None,
+            };
+
+            Some(feature_collection)
+        }
+        _ => None,
+    }
+}
+
+fn geojson_feature_point(stop_id: &String, gtfs: &Gtfs) -> Option<Feature> {
+    gtfs.stops.get(stop_id).map(|stop| {
+        let stop_geom = get_stop_geom(stop);
+        let mut properties = Map::new();
+
+        properties.insert(String::from("id"), to_value(&stop.id).unwrap());
```

Comment:
So this one is tricky, and I blame `geojson`’s documentation. `unwrap` are generally a big big no.
If something is wrong, the whole program will stop. Errors should be handled, no ignored.

However, you know here that you have `String`, so it can’t fail in many situations, so you can keep as such and add a comment justifying why you are introducing unsafe code.

Because of Murphy’s law, it will someday fail because the string is to big or something like that. To handle that, I’d build a function something like (not tested):

```rust
fn stop_properties(stop: &Stop) -> Result<serde_json::Map, anyhow::Error> {
  let mut properties = Map::new();
  properties.insert("id".to_string, to_value(&stop.id))?;
  properties.insert("name".to_string(), to_value(&stop.name))?;
  properties
}

/// and in geojson_feature_point
        Feature {
            geometry: get_stop_geom(stop),
            bbox: None,
            properties: stop_properties(stop).ok(), // It will be None if it returned an error
            id: None,
            foreign_members: None,
        }

```

Reply:
I have replaced unwrap() by unwrap_or() with an empty default value. I think it works and still gives some properties, if for example "name" fails to be converted, but "id" is ok.